dvc/state.py

```diff
@@ -449,6 +450,10 @@ def remove_unused_links(self, used):
                 remove(path)
                 unused.append(relpath)
 
-        for relpath in unused:
-            cmd = "DELETE FROM {} WHERE path = ?".format(self.LINK_STATE_TABLE)
-            self._execute(cmd, (relpath,))
+        for chunk_unused in to_chunks(
+            unused, chunk_size=SQLITE_MAX_VARIABLES_NUMBER
+        ):
+            cmd = "DELETE FROM {} WHERE path IN ({})".format(
```

Comment:
So MAX_PATH is ~4096 usually. so MAX_PATH * SQLITE_MAX_VARIABLES_NUMBER ~ 4M, and SQLITE_MAX_SQL_LENGTH ~ 1M. Theoretically, there might be a problem. But those numbers of files are usually in directories, for which we only add one entry - path to the directory itself, so it is unlikely that we will hit this, but if we do then sql will raise SQLITE_TOOBIG, and we can deal with it later.